plugins/org.yakindu.sct.generator.c/src/org/yakindu/sct/generator/c/CExpressionsGenerator.xtend

```diff
@@ -165,14 +168,19 @@ class CExpressionsGenerator extends ExpressionsGenerator {
 
 	def dispatch CharSequence code(FeatureCall it, Operation target) {
 		if (target.eContainer instanceof ComplexType) {
-			return '''«target.getFunctionId(owner.featureOrReference)»(«owner.getHandle(scHandle + "->")»)'''
+			return '''«target.getFunctionId(owner.featureOrReference)»(«owner.getHandle(scHandle + "->")»«FOR arg : expressions BEFORE ', ' SEPARATOR ', '»«arg.code»«ENDFOR»)'''
 		}
 		'''«it.owner.code».«target.access»(«FOR arg : expressions SEPARATOR ', '»«arg.code»«ENDFOR»)'''
 	}
 
 	def dispatch CharSequence code(FeatureCall it, Property target) '''«it.owner.code»«IF !(target.eContainer instanceof ComplexType)».«target.access»«ENDIF»'''
 
-	def dispatch CharSequence code(FeatureCall it, Enumerator target) '''«target.access»'''
+	def dispatch CharSequence code(FeatureCall it, Enumerator target) {
+		if(!target.eContainer.originTraces.nullOrEmpty) {
```

Comment:
With PR #2912 this can be simplified to `target.eContainer.isStateEnum`

Reply:
:+1: